src/Sentry/Laravel/Tracing/Storage/TracingFilesystem.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace Sentry\Laravel\Tracing\Storage;
+
+use Exception;
+use Illuminate\Contracts\Filesystem\Cloud as CloudFilesystemContract;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Filesystem\FilesystemAdapter;
+use Sentry\Tracing\SpanContext;
+use function Sentry\trace;
+
+/**
+ * Decorates the underlying filesystem by wrapping all calls to it with tracing.
+ *
+ * Parameters such as paths, directories or options are attached to the span as data,
+ * parameters that contain file contents are omitted due to potential problems with
+ * payload size or sensitive data.
+ */
+class TracingFilesystem implements CloudFilesystemContract
+{
+    /** @var Filesystem */
+    protected $filesystem;
+
+    public function __construct(Filesystem $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    /**
+     * @param list<mixed> $args
+     * @param array<string, mixed> $data
+     */
+    protected function withTracing(string $method, array $args, array $data)
+    {
+        $context = new SpanContext();
+        $context->setOp("file.{$method}"); // See https://develop.sentry.dev/sdk/performance/span-operations/#web-server
```

Comment:
It's actually under mobile, let's keep it generic.

```suggestion
        $context->setOp("file.{$method}"); // See https://develop.sentry.dev/sdk/performance/span-operations/
```

Reply:
It is both under Web Server and Mobile, so I think we can be specific here.

Reply:
Woops, yes. I searched for `file.` 🤦 

Nothing was said!